Source/ACE/Entity/Landblock.cs

```diff
@@ -843,6 +843,17 @@ private void ItemTransferContainerInternal(ObjectGuid woGuid, Container containe
             wo.Location = null;
             wo.InitializeAceObjectForSave();
             container.AddToInventory(wo, placement);
+
+            // Was Item controlled by a generator?
+
+            if (wo.GeneratorId > 0)
```

Comment:
Should this be a function of the landblock, or a function of the item?

In other words, is it the landblock's responsibility to know that generators exist, and that objects may have generators, or is it the object's responsibility to notify the generator on pickup.  (And the landblock's responsibility to notify the item it was picked up).

IMO landblocks should have objects and be oblivious to what they are, or if they have generators.

Reply:
More thinking to be done here, a TODO perhaps?